### PR TITLE
Feature/user controller endpoint

### LINKED ISSUE #5: Create endpoint to receive Google User profile from the Frontend

### PR DESCRIPTION
**Pull request by rarques**
Closes #5

### CODE REVIEW

File: src/main/java/com/garannodou/role/controller/GoogleAuthRequest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    private GoogleBasicProfile googleBasicProfile;
+    @NotNull
+    private GoogleAuthResponse googleAuthResponse;
+
+    public GoogleAuthRequest() {
+    }
+
+    @Getter
+    @Setter
+    static class GoogleBasicProfile {
```
Comment (rarques) [inline src/main/java/com/garannodou/role/controller/GoogleAuthRequest.java]: "static" may not be needed?

File: src/main/java/com/garannodou/role/controller/GoogleAuthRequest.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        @NotNull private String imageUrl;
+        @NotNull private String email;
+
+        public GoogleBasicProfile() {
+        }
+    }
+
+    @Getter
+    @Setter
+    static class GoogleAuthResponse {
```
Comment (rarques) [inline src/main/java/com/garannodou/role/controller/GoogleAuthRequest.java]: Same as above

File: src/main/java/com/garannodou/role/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+public class UserController {
+
+    private GoogleUserAuthenticationService authenticationService;
+
+    public UserController(GoogleUserAuthenticationService authenticationService) {
+        this.authenticationService = authenticationService;
+    }
+
+    @PostMapping("/authentication")
+    public ResponseEntity<String> authenticateGoogleUser(@RequestBody GoogleAuthRequest request) {
```
Comment (rarques) [inline src/main/java/com/garannodou/role/controller/UserController.java]: Add "@Valid" to the request body

File: src/test/java/com/garannodou/role/controller/UserControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+import org.springframework.test.web.servlet.MockMvc;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.*;
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+
+
+@ExtendWith(SpringExtension.class)
```
Comment (rarques) [inline src/test/java/com/garannodou/role/controller/UserControllerTest.java]: Still needed using MockMvc?

File: src/test/java/com/garannodou/role/controller/UserControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+@WebMvcTest(UserController.class)
+class UserControllerTest {
+
+    // TODO: Input validation cases
+    // TODO: Call service and return 201
```
Comment (rarques) [inline src/test/java/com/garannodou/role/controller/UserControllerTest.java]: Remove TODO

File: src/main/java/com/garannodou/role/controller/UserController.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        String response = authenticationService.authenticate(request);
+
+        return new ResponseEntity<>(response, HttpStatus.CREATED);
```
Comment (mduenast) [inline src/main/java/com/garannodou/role/controller/UserController.java:25]: specifies further the type of return

File: src/main/java/com/garannodou/role/controller/GoogleAuthRequest.java (hunk 1 above)
Comment (rarques) [reply src/main/java/com/garannodou/role/controller/GoogleAuthRequest.java]: "static" not needed. GoogleBasicProfile and GoogleAuth only make sense to exist within GoogleAuthRequest.

File: src/main/java/com/garannodou/role/controller/GoogleAuthRequest.java (hunk 2 above)
Comment (rarques) [reply src/main/java/com/garannodou/role/controller/GoogleAuthRequest.java]: https://github.com/garan-no-dou/role-ws/pull/12#discussion_r479754467

**Review by mduenast — APPROVED** [review]

**Pull request merged**